### PR TITLE
fix for https://github.com/inexorabletash/polyfill/issues/66

### PR DESCRIPTION
**Pull request by Yaffle**
https://github.com/inexorabletash/polyfill/issues/66

### CODE REVIEW

File: typedarray.js [hunk 1]
```diff
@@ -177,7 +177,16 @@
 
       if (v >= pow(2, 1 - bias)) {
         e = min(floor(log(v) / LN2), 1023);
-        f = roundToEven(v / pow(2, e) * pow(2, fbits));
+        var significand = v / pow(2, e);
+        if (significand < 1) {
+          e -= 1;
+          significand *= 2;
+        }
+        if (significand >= 2) {
```
Comment (Yaffle) [inline typedarray.js]: Test case:
```
var x = new Float64Array(1), y = Math.pow(2, -901) * (1 + 2 * Number.EPSILON);
x[0] = y;
console.log(x[0] - y);
```

File: typedarray.js [hunk 2]
```diff
@@ -177,7 +177,16 @@
 
       if (v >= pow(2, 1 - bias)) {
         e = min(floor(log(v) / LN2), 1023);
-        f = roundToEven(v / pow(2, e) * pow(2, fbits));
+        var significand = v / pow(2, e);
+        if (significand < 1) {
+          e -= 1;
```
Comment (Yaffle) [inline typedarray.js]: Test case (from the #66):
```
var x = new Float64Array(1), y = Number.MAX_VALUE / 2;
x[0] = y;
console.log(x[0] - y);
```

File: typedarray.js [hunk 3]
```diff
@@ -177,7 +177,16 @@
 
       if (v >= pow(2, 1 - bias)) {
         e = min(floor(log(v) / LN2), 1023);
-        f = roundToEven(v / pow(2, e) * pow(2, fbits));
+        var significand = v / pow(2, e);
+        if (significand < 1) {
```
Comment (Yaffle) [inline typedarray.js]: Test case (from the #66):
```
var x = new Float64Array(1), y = Number.MAX_VALUE / 2;
x[0] = y;
console.log(x[0] - y);
```

**Comment by inexorabletash** [comment]
Thanks for the fix!

**Pull request merged**